wagl/acquisition/landsat.py

```diff
@@ -144,6 +150,95 @@ def __init__(self, pathname, uri, acquisition_datetime, band_name='BAND 1',
         self._classification_type = 'U'
         self._international_designator = '99020A'
 
+        self._gap_mask = None
+
+    def _extract_gap_mask(self):
+        """
+        Extract the gap mask from within the tar.
+        The gap mask is a TIF that is then pushed through gzip,
+        horrible, and in order to read it, we need to unpack from
+        the tar, decompress using gzip, then write to disk so GDAL
+        can read it.
+        """
+        # mask files are contained in a sub-directory named 'gap-mask'
+        path = Path(self.uri.replace('!', '')[6:])
```

Comment:
This line isn't completely clear to me: is it assuming a tar uri, and the "6" offset is length of '`tar://`'? It's could quietly have strange behaviour if another uri scheme is passed in (like `tar+gz://` or `zip://` or `file://`). I'm not sure how much that matters.

Reply:
True, it could be problematic for a different scheme. The current driver implementation for Landsat will handle a directory, or a tar file.  The tar file could be '`.tar`' or '`.tar.gz`', so we're not being explicit on the scheme to require '`tar+gz`'.  We're leaving it up to the tar module to unpack, as well as rasterio.
We will get into some trouble if we process a directory of a landsat-7 acquisition, but the USGS spec is a '`.tar.gz`', so we shouldn't run into issues. If USGS updates the package specification, then we'll update the driver to handle that change.
So I'd say for the time being, it is not much of an issue, but I agree that we could do something to handle that case better.

Reply:
Just tested a version that includes a check for a '`tar`' scheme and raises a `NotImplementedError`.  Unfortunately, other tests for Landsat-7 acquisitions packaged as directories now fail.
These tests, whilst old, are still relevant and changing more tests to no longer handle directories is more time consuming I think it is worth trying to resolve (for the time being).
We need to get processing underway, so we'll have to go with as-is and simply take more care if USGS update their package specification.